Add section showing how to enable a delegator factory

settermjd opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

While the docs are pretty complete up until this point, it doesn't contain the final step, which shows how to enable the delegator factory. This change adds that missing step.

settermjd committed:
Add section showing how to enable a delegator factor

While the docs are pretty complete up until this point, it doesn't
contain the final step, which shows how to enable the delegator factory.
This change adds that missing step.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd commented:
@froschdesign, the previous PR on the correct branch.

froschdesign commented on docs/book/v3/features/error-handling.md:
````diff
@@ -213,6 +213,22 @@ class LoggingErrorListenerDelegatorFactory
 }
 ```
 
+Then, enable the delegator factory by registering it in the `delegators` element of the relevant [ConfigProvider class](https://docs.laminas.dev/laminas-config-aggregator/config-providers/), as in the example below.
````
What I don't like on the entire page: listeners should be created, configuration must be extended and now a config provider is to be extended. But where is the description in which file and under which path this should be done?
While this may be flexible and different for each application, an example must be given.

Otherwise we say: _bird eat and die anyway._

settermjd replied:
I don't quite follow the sentence at the end, however, I appreciate your point. I can either work in the required changes in this PR, or in a subsequent one. Which would you prefer?

froschdesign replied:
Add a filename and path to your code example, everything else can be done in another pull request.

Example: https://docs.laminas.dev/laminas-form/v3/application-integration/usage-in-a-laminas-mvc-application/#register-form-and-controller

settermjd replied:
> Add a filename and path to your code example, everything else can be done in another pull request.
> 
> Example: https://docs.laminas.dev/laminas-form/v3/application-integration/usage-in-a-laminas-mvc-application/#register-form-and-controller

That's an excellent example. Thanks, @froschdesign. Will do that a little later, this morning.

settermjd replied:
Hey @froschdesign, I added the example that I thought might most often come to mind, that being the `getDependencies` method in _src/App/src/ConfigProvider.php_. The `App` module is the one that comes with the Mezzio Skeleton project and used most often in the docs, iirc.

I italicised the path as I picked up that convention while working at Twilio. It nicely separates files from code, which I hope makes the docs that much easier to scan/read.

settermjd committed:
Extend the delegator factor example

This change provides an an example file and method where the change
could happen, one that should help especially new developers to this
concept. For more experienced developers, they likely already know what
to do, and may do it in a different way. But for less-experienced
developers, or ones seeing the concept for the first time, this should
help avoid any confusion or uncertainty.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

froschdesign commented on docs/book/v3/features/error-handling.md:
````diff
@@ -213,6 +213,23 @@ class LoggingErrorListenerDelegatorFactory
 ```
 
+Then, enable the delegator factory by registering it in the `delegators` element of the relevant [ConfigProvider class](https://docs.laminas.dev/laminas-config-aggregator/config-providers/).
+For example, in _src/App/src/ConfigProvider.php_, update the `getDependencies()` method as follows:
````
```suggestion
For example, in `src/App/src/ConfigProvider.php`, update the `getDependencies()` method as follows:
```

settermjd replied:
As mentioned in https://github.com/mezzio/mezzio/pull/116#discussion_r893430990, I italicised the path as I picked up that convention while working at Twilio. It nicely separates files from code, which I hope makes the docs that much easier to scan/read. Would you mind me making this formatting change?

froschdesign replied:
> I italicised the path as I picked up that convention while working at Twilio. It nicely separates files from code, which I hope makes the docs that much easier to scan/read.

This means that all file names and paths in the entire documentation will look different than on this page.

froschdesign replied:
Please don't get me wrong, better readability is always welcome, but don't see a chance to change the convention at this point.

settermjd replied:
While I love the italicised approach, I accept your wisdom.

settermjd committed:
Update docs/book/v3/features/error-handling.md

Revert italicisation to inline code block for consistency.

Co-authored-by: Frank Brückner <info@froschdesignstudio.de>
Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

froschdesign commented on docs/book/v3/features/error-handling.md:
```diff
@@ -213,6 +213,23 @@ class LoggingErrorListenerDelegatorFactory
+{
+    return [
+        'delegators' => [
+            ErrorHandler::class => [
+                LoggingErrorListenerDelegatorFactory::class
```
```suggestion
            Laminas\Stratigility\Middleware\ErrorHandler::class => [
                Acme\LoggingErrorListenerDelegatorFactory::class
```

settermjd replied:
I agree with this suggestion, but before committing it, is Acme the standard term used in the Mezzio docs for a hypothetical module? If so, then I should change the file path that I used in this PR as well.

froschdesign replied:
> …is Acme the standard term used in the Mezzio docs for a hypothetical module?

Only in the documentation of Mezzio itself: https://github.com/search?l=Markdown&q=org%3Amezzio+Acme&type=Code
Therefore, it is correct to modify the path name.

(But in the long run, a better name / example must be found.)

gsteel commented:
@settermjd Was this handled in #181 ?? If so, can you close this patch. Cheers.
